packages/react/dismissable-layer/src/DismissableLayer.tsx

```diff
@@ -301,31 +302,38 @@ function useFocusOutside(onFocusOutside?: (event: FocusOutsideEvent) => void) {
  * -----------------------------------------------------------------------------------------------*/
 
 function createTotalLayerCount(displayName?: string) {
-  const TotalLayerCountContext = React.createContext<{
-    total: number;
-    setTotal: React.Dispatch<React.SetStateAction<number>>;
-  }>({ total: 0, setTotal: () => {} });
+  const [TotalLayerCountProviderImpl, useTotalLayerCountContext] = createContext(
+    'TotalLayerCount',
+    { total: 0, onTotalIncrease: () => {}, onTotalDecrease: () => {} }
+  );
 
   const TotalLayerCountProvider: React.FC = ({ children }) => {
     const [total, setTotal] = React.useState(0);
-    const context = React.useMemo(() => ({ total, setTotal }), [total, setTotal]);
     return (
-      <TotalLayerCountContext.Provider value={context}>{children}</TotalLayerCountContext.Provider>
+      <TotalLayerCountProviderImpl
+        total={total}
+        onTotalIncrease={React.useCallback(() => setTotal((n) => n + 1), [])}
```

Comment:
nice refactoring to align with other places 👍

Reply:
Mostly did it because it made the `defaultContext` easier to type 😅 nice win in the process.